### DISCUSSION TITLE
Is there a way nerdctl logs to show logs from containers started by kubelet.

### DISCUSSION THREAD

**Discussion by jeffgtxjava** [containerd/nerdctl#1203] [Q&A]
Based on the FAQ's [here](https://github.com/containerd/nerdctl/blob/master/docs/faq.md#how-is-nerdctl-different-from-ctr-and-crictl-) I assumed nerdctl would be able to fetch the logs from the containers started by kubelet.

Any specific reason why it wont show `logs` while it can `ps`, `inspect` and even `exec`

### Logs
```
-sh-4.2# nerdctl logs 92b39c8386ef 
WARN[0000] Currently, `nerdctl logs` only supports containers created with `nerdctl run -d` 
FATA[0000] open /cowdata/nerd-bootstrap/29c6aec2/containers/92b39c8386ef758e2327a26c1070771c6d91f9365f74a4241f691f1327267c0c/log-config.json: no such file or directory 
-sh-4.2# nerdctl inspect 92b39c8386ef | wc -l
89
```

### Inspect
```
-sh-4.2# nerdctl inspect 92b39c8386ef | wc -l
89
```

### Exec
```
-sh-4.2# nerdctl exec -it 92b39c8386ef date
Wed Jul  6 12:25:42 GMT 2022
```

**NOTE**: `crictl` is able to do the same.

```
-sh-4.2# crictl logs 92b39c8386ef &> /tmp/testlogs

-sh-4.2# wc -l /tmp/testlogs 
24 /tmp/testlogs
-sh-4.2# 
```

**Comment by AkihiroSuda** [comment]
> Any specific reason why it wont show logs while it can ps, inspect and even exec

Because containerd still lacks high-level logging infrastructure.
nerdctl implements its own logging drivers (`json-file`, `journald`, `fluentd`) to emulate the UX of Docker, while the CRI implements its own logging driver.

**Reply by jeffgtxjava** [reply]
@AkihiroSuda Just trying to understand more. Apologies for my tech ignorance. Isn't crictl logs also operating on top of same containerd. I was under assumption we can create a task on containerd to get the logs.

**Reply by AkihiroSuda** [reply]
Same daemon, but different logging infra.

If we want to let nerdctl read CRI logs we will have to duplicate the logic from https://github.com/kubernetes/kubernetes/blob/v1.24.2/pkg/kubelet/kuberuntime/logs/logs.go

**Reply by jeffgtxjava** [reply]
@AkihiroSuda Should we create a issue so it would eventually get picked up as a feature request?

I feel this would be a great addition as we are able to perform `inspect, exec` and multiple other operations on the containers started by kubelet and would help ppl migrate from docker post docker-shim deprecation.

**Reply by AkihiroSuda** [reply]
Yes, please

**Reply by fahedouch** [reply]
@jeffgtxjava we have a ticket that raise logging driver implen we can ask for implementation here https://github.com/containerd/nerdctl/issues/1039
